src/main/java/atmosfears/AtmosFearsBE/service/Predictor.java

```diff
@@ -52,15 +59,16 @@ public static double[] predict(String csv_file_path){
             // RETRIEVING RESULT
             rCaller.setRCode(code);
             rCaller.runAndReturnResult("out");
-            return rCaller.getParser().getAsDoubleArray("_pred");
+            return DoubleStream.of( rCaller.getParser().getAsDoubleArray("_pred") ).boxed().collect(Collectors.toCollection(ArrayList::new));
+
 
         }
         catch (Exception e) {
             Logger.getLogger(examples.Main.class.getName()).log(Level.SEVERE, e.getMessage());
         }
 
         // RETURN -1 IN CASE OF ERROR
-        return new double[]{-1};
+        return new ArrayList<>(Arrays.asList(-1.0));
```

Comment:
also not sure about that, why we want to return [-1] if error occured?

Reply:
Sorry, that's a habit from my job. I'll set it to return null.